### PR TITLE
cgaustin's mods to sensor.py for dswe and lst

### PR DESCRIPTION
**Pull request by caustin-usgs**
updates to sensor.py to handle dswe and lst availability

### CODE REVIEW

**Comment by sfoga-usgs** [comment]
Looks like you're missing the brightness temperature product option, which
applies to all sensor classes except 'LandsatOLI'.

On Thu, Jan 21, 2016 at 9:56 AM, Clay Austin <notifications@github.com>
wrote:

> updates to sensor.py to handle dswe and lst availability
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/USGS-EROS/espa-api/pull/2
> Commit Summary
>
>    - reorg landsat 4, 5 and 7 class defs in sensor.py
>    - sensor.py mods
>    - updates to sensor.py to handle dswe and lst availability
>
> File Changes
>
>    - *M* api/domain/sensor.py
>    <https://github.com/USGS-EROS/espa-api/pull/2/files#diff-0> (123)
>
> Patch Links:
>
>    - https://github.com/USGS-EROS/espa-api/pull/2.patch
>    - https://github.com/USGS-EROS/espa-api/pull/2.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/USGS-EROS/espa-api/pull/2>.
>



-- 
Steven Foga
Remote Sensing Scientist
Stinger Ghaffarian Technologies (SGT, Inc.)
Contractor to the U.S. Geological Survey (USGS)
Earth Resources Observation and Science (EROS) Center
47914 252nd Street
Sioux Falls, SD  57198
(605) 594-2839
sfoga@usgs.gov

**Comment by caustin-usgs** [comment]
@sfoga-usgs whats the product code for the brightness temperature product ?

File: api/domain/sensor.py [hunk 1]
```diff
@@ -327,19 +388,19 @@ def instance(product_id):
 
     instances = {
         'tm4': (r'^lt4\d{3}\d{3}\d{4}\d{3}[a-z]{3}[a-z0-9]{2}$',
```
Comment (rdilley-usgs) [inline api/domain/sensor.py]: When this is finished, will we only be processing collection based data??
Either way, these regular expression tests need to be modified to accomodate collection IDs.

**Comment by sfoga-usgs** [comment]
@caustin-usgs on the web side, it's "include_sr_thermal", so I'm guessing it would be "sr_thermal" in your code.

File: api/domain/sensor.py (hunk 1 above)
Comment (klsmith-usgs) [inline api/domain/sensor.py]: They would be expanded upon to accommodate the different collections

Comment (dhill-usgs) [inline api/domain/sensor.py]: not sr thermal in the new code, thermal is not derived from an sr product.
It comes straight from level 1, right Ron?

On Thu, Jan 21, 2016 at 10:21 AM, Kelcy Smith <notifications@github.com>
wrote:

> In api/domain/sensor.py
> <https://github.com/USGS-EROS/espa-api/pull/2#discussion_r50423082>:
>
> > @@ -327,19 +388,19 @@ def instance(product_id):
> >
> >      instances = {
> >          'tm4': (r'^lt4\d{3}\d{3}\d{4}\d{3}[a-z]{3}[a-z0-9]{2}$',
>
> They would be expanded upon to accommodate the different collections
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/USGS-EROS/espa-api/pull/2/files#r50423082>.
>



-- 
David V. Hill
LSRD
SGT
Contractor to USGS Earth Resources Observation And Science Center (EROS)
47914 252nd Street
Sioux Falls, SD 57198-9801
605.594.2639
dhill@usgs.gov

**Comment by klsmith-usgs** [comment]
Terrible naming scheme if you ask me. When I think of SR Thermal, I would think of land surface temperature (LST). Nevermind the fact that it isn't a reflectance attribute, but the fact that it is coupled with an at surface product. Brightness temperature to me would be more like a TOA product.
At least that's what I would think coming into these products blind.

**Comment by rdilley-usgs** [comment]
Both TOA and BT are derived form the Level 1 input data.
So tm_toa, tm_bt, etm_toa, etm_bt, tirs_bt, oli_toa, olitirs_toa, olitirs_bt

I don't really like OLITIRS being classified as a  as a sensor, when it is two, but that is an input product to us so we have to live with it.

**Comment by caustin-usgs** [comment]
@rdilley-usgs TOA?

**Comment by dhill-usgs** [comment]
sensor_productlevel_band

etm|tm|mss|oli|tirs|olitirs _ l1|toa|sr _ ndvi|ndmi|nbr|nbr2|evi|savi|msavi

On Thu, Jan 21, 2016 at 12:32 PM, Clay Austin <notifications@github.com>
wrote:

> @rdilley-usgs <https://github.com/rdilley-usgs> TOA?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/USGS-EROS/espa-api/pull/2#issuecomment-173666229>.
>



-- 
David V. Hill
LSRD
SGT
Contractor to USGS Earth Resources Observation And Science Center (EROS)
47914 252nd Street
Sioux Falls, SD 57198-9801
605.594.2639
dhill@usgs.gov

**Comment by dhill-usgs** [comment]
etm|tm|mss|oli|tirs|olitirs|mod09|myd09|mod13|myd13 _ l1|toa|sr _
ndvi|ndmi|nbr|nbr2|evi|savi|msavi

On Thu, Jan 21, 2016 at 12:34 PM, Hill, David <dhill@usgs.gov> wrote:

> sensor_productlevel_band
>
> etm|tm|mss|oli|tirs|olitirs _ l1|toa|sr _ ndvi|ndmi|nbr|nbr2|evi|savi|msavi
>
> On Thu, Jan 21, 2016 at 12:32 PM, Clay Austin <notifications@github.com>
> wrote:
>
>> @rdilley-usgs <https://github.com/rdilley-usgs> TOA?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/USGS-EROS/espa-api/pull/2#issuecomment-173666229>.
>>
>
>
>
> --
> David V. Hill
> LSRD
> SGT
> Contractor to USGS Earth Resources Observation And Science Center (EROS)
> 47914 252nd Street
> Sioux Falls, SD 57198-9801
> 605.594.2639
> dhill@usgs.gov
>



-- 
David V. Hill
LSRD
SGT
Contractor to USGS Earth Resources Observation And Science Center (EROS)
47914 252nd Street
Sioux Falls, SD 57198-9801
605.594.2639
dhill@usgs.gov

**Comment by dhill-usgs** [comment]
sorry, Top of Atmosphere.

On Thu, Jan 21, 2016 at 12:36 PM, Hill, David <dhill@usgs.gov> wrote:

> etm|tm|mss|oli|tirs|olitirs|mod09|myd09|mod13|myd13 _ l1|toa|sr _
> ndvi|ndmi|nbr|nbr2|evi|savi|msavi
>
> On Thu, Jan 21, 2016 at 12:34 PM, Hill, David <dhill@usgs.gov> wrote:
>
>> sensor_productlevel_band
>>
>> etm|tm|mss|oli|tirs|olitirs _ l1|toa|sr _
>> ndvi|ndmi|nbr|nbr2|evi|savi|msavi
>>
>> On Thu, Jan 21, 2016 at 12:32 PM, Clay Austin <notifications@github.com>
>> wrote:
>>
>>> @rdilley-usgs <https://github.com/rdilley-usgs> TOA?
>>>
>>> —
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/USGS-EROS/espa-api/pull/2#issuecomment-173666229>.
>>>
>>
>>
>>
>> --
>> David V. Hill
>> LSRD
>> SGT
>> Contractor to USGS Earth Resources Observation And Science Center (EROS)
>> 47914 252nd Street
>> Sioux Falls, SD 57198-9801
>> 605.594.2639
>> dhill@usgs.gov
>>
>
>
>
> --
> David V. Hill
> LSRD
> SGT
> Contractor to USGS Earth Resources Observation And Science Center (EROS)
> 47914 252nd Street
> Sioux Falls, SD 57198-9801
> 605.594.2639
> dhill@usgs.gov
>



-- 
David V. Hill
LSRD
SGT
Contractor to USGS Earth Resources Observation And Science Center (EROS)
47914 252nd Street
Sioux Falls, SD 57198-9801
605.594.2639
dhill@usgs.gov

File: api/domain/sensor.py [hunk 2]
```diff
@@ -232,38 +232,27 @@ def __init__(self, product_id):
...
-class LandsatTM4(Landsat):
-    """Models Thematic Mapper based products"""
-    products = ['tm_sr', 'tm_toa', 'tm_l1',
+class LandsatTM(Landsat):
+    """Models Landsat TM only products"""
+    products = ['tm_sr', 'tm_toa', 'tm_l1', 'tm_dswe',
                 'tm_sr_ndvi', 'tm_sr_ndmi', 'tm_sr_evi',
                 'tm_sr_savi', 'tm_sr_msavi', 'tm_sr_nbr',
-                'tm_sr_nbr2', 'source', 'source_metadata']
+                'tm_sr_nbr2', 'tm_lst', 'tm_bt', 'source',
```
Comment (dhill-usgs) [inline api/domain/sensor.py]: remove references to 'source'.  That option was removed from the system a month or two ago.  'source_metadata' is still valid.

File: api/domain/sensor.py [hunk 3]
```diff
@@ -272,14 +261,13 @@ def __init__(self, product_id):
...
 class LandsatOLITIRS(Landsat):
-    """Models combined Landsat 8 OLI/TIRS products"""
-
+    """Models Landsat OLI/TIRS only products"""
     products = ['olitirs_sr', 'olitirs_toa', 'olitirs_l1',
                 'olitirs_sr_ndvi', 'olitirs_sr_ndmi',
                 'olitirs_sr_evi', 'olitirs_sr_savi',
-                'olitirs_sr_msavi', 'olitirs_sr_nbr',
+                'olitirs_sr_msavi', 'olitirs_sr_nbr', 'olitirs_bt',
                 'olitirs_sr_nbr2', 'source', 'source_metadata']
```
Comment (dhill-usgs) [inline api/domain/sensor.py]: we're still missing products/bands.  We need to sit down in a room with a board and capture the full suite.  Ron needs to be there.  I'm going to print this off then go over it with him so we don't keep going over this

**Pull request merged**